Src/Xer.Cqrs.Extensions.SimpleInjector/CqrsBuilder.cs

```diff
@@ -0,0 +1,104 @@
+    {
+        private readonly Container _container;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly CqrsCommandHandlerSelector _commandHandlerSelector;
+        private readonly CqrsEventHandlerSelector _eventHandlerSelector;
+
+        internal CqrsBuilder(Container container)
+        {
+            _container = container;
+            // Cast to IServiceProvider so that container will not throw if type is not registered.
+            _serviceProvider = container;
+            _commandHandlerSelector = new CqrsCommandHandlerSelector(container);
+            _eventHandlerSelector = new CqrsEventHandlerSelector(container);
+        }
+
+        public ICqrsBuilder RegisterCommandHandlers(Action<ICqrsCommandHandlerSelector> selector)
+        {
+            if (selector == null)
+            {
+                throw new ArgumentNullException(nameof(selector));
+            }
+
+            selector.Invoke(_commandHandlerSelector);
+
+            _container.RegisterSingleton<CommandDelegator>(() =>
+            {
+                if (tryGetInstance(out IEnumerable<CommandHandlerDelegateResolver> commandHandlerResolvers))
+                {
+                    CommandHandlerDelegateResolver[] resolverArray = commandHandlerResolvers.ToArray();
+
+                    if (resolverArray.Length == 1)
+                    {
+                        return new CommandDelegator(resolverArray[0]);
+                    }
+
+                    return new CommandDelegator(CompositeMessageHandlerResolver.Compose(resolverArray));
+                }
+                
+                return new CommandDelegator(new SingleMessageHandlerRegistration().BuildMessageHandlerResolver());
+            });
+
+            return this;
+        }
+
+        public ICqrsBuilder RegisterEventHandlers(Action<ICqrsEventHandlerSelector> selector)
+        {
+            if (selector == null)
+            {
+                throw new ArgumentNullException(nameof(selector));
+            }
+            
+            selector(_eventHandlerSelector);
+
+            _container.RegisterSingleton<EventDelegator>(() =>
+            {                
+                if (tryGetInstance(out IEnumerable<EventHandlerDelegateResolver> eventHandlerResolvers))
+                {
+                    EventHandlerDelegateResolver[] resolverArray = eventHandlerResolvers.ToArray();
+
+                    if (resolverArray.Length == 1)
+                    {
+                        return new EventDelegator(resolverArray[0]);
+                    }
+
+                    return new EventDelegator(CompositeMessageHandlerResolver.Compose(resolverArray));
+                }
+
+                return new EventDelegator(new SingleMessageHandlerRegistration().BuildMessageHandlerResolver());
+            });
+
+            return this;
+        }
+
+        private bool tryGetInstance<T>(out T instance) where T : class
```

Comment:
Use the default guidelines `TryGetInstance`

Reply:
Oh, shouldn't private functions be camel-cased?

Reply:
I always follow this guideline https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines. It has no distinction between different methods (private, internal or public etc). I am in favour of following those guidelines.